### PR TITLE
Exposing accept-options and connect-options in the management layer...

### PR DESCRIPTION
**Pull request by krismcqueen**
...in a flexible manner to support extensibility.

### CODE REVIEW

**Comment by krismcqueen** [comment]
@jitsni any feedback would be appreciated, especially in how to better manage the strings from the config as related to the strings in the options Map.

File: server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultServiceContext.java [hunk 1]
```diff
@@ -282,9 +287,18 @@ public boolean equals(Object otherObject) {
 
     @Override
     public int hashCode() {
-        int result = serviceType != null ? serviceType.hashCode() : 0;
-        result = 31 * result + (accepts != null ? accepts.hashCode() : 0);
-        return result;
+        if (hashCode == -1) {
+            int result = serviceType != null ? serviceType.hashCode() : 0;
+            result = 31 * result + (accepts != null ? accepts.hashCode() : 0);
```
Comment (jitsni) [inline server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultServiceContext.java]: Objects.hashCode(accepts) should correspond to (accepts != null ? accepts.hashCode() : 0)
Perhaps even better, why cannot we use Objects.hash(serviceType, accepts, serviceName)

File: management/src/main/java/org/kaazing/gateway/management/config/ServiceDefaultsConfigurationBeanImpl.java [hunk 2]
```diff
@@ -95,35 +99,120 @@ public String getAcceptOptions() {
                     jsonOptions.put("ssl.verify-client", "none");
                 }
 
-                jsonOptions.put("ws.maximum.message.size", acceptOptions.get("ws.maximum.message.size"));
+                jsonOptions.put("ws.maximum.message.size", acceptOptions.remove("ws.maxMessageSize"));
```
Comment (jitsni) [inline management/src/main/java/org/kaazing/gateway/management/config/ServiceDefaultsConfigurationBeanImpl.java]: ws.maximum.message.size seems to be expanded into ws.maxMessageSize, ws[ws/rfc6455].ws[ws/rfc6455].maxMessageSize, ws[ws/draft-7x].ws[ws/draft-7x].maxMessageSize in DefaultAcceptOptionsContext

Do we need to put all three values ? How do we handle such cases ?

**Comment by jitsni** [comment]
@dpwspoon can you also review here (I don't have full understanding here)

File: management/src/main/java/org/kaazing/gateway/management/config/ServiceDefaultsConfigurationBeanImpl.java (hunk 2 above)
Comment (krismcqueen) [inline management/src/main/java/org/kaazing/gateway/management/config/ServiceDefaultsConfigurationBeanImpl.java]: They are explicitly ignored before processing the rest of the options.  See the loop towards the bottom of the method.

**Comment by krismcqueen** [comment]
@jitsni code has been updated as per your feedback.  Let me know if there is anything else.

**Pull request merged**